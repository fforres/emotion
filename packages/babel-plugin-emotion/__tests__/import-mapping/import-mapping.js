import babelTester from 'babel-tester'
import plugin from 'babel-plugin-emotion'

babelTester('import mapping', __dirname, {
  plugins: [
    [
      plugin,
      {
        importMap: {
          'package-one': {
            nonDefaultStyled: {
              canonicalImport: ['@emotion/styled', 'default']
            },
            someCssFromCore: {
              canonicalImport: ['@emotion/core', 'css']
            }
          },
          'package-two': {},

          'package-three': {},
          'package-four': {}
        }
      }
    ]
  ]
})
