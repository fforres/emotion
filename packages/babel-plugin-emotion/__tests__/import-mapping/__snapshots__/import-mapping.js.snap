// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import mapping non-default-styled 1`] = `
"import { nonDefaultStyled } from 'package-one'

let SomeComp = nonDefaultStyled.div({ color: 'hotpink' })


      ↓ ↓ ↓ ↓ ↓ ↓

import { nonDefaultStyled as _styled } from \\"package-one\\";

let SomeComp = _styled(\\"div\\", {
  target: \\"e1nqcgbi0\\",
  label: \\"SomeComp\\"
})(process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
} : {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIm5vbi1kZWZhdWx0LXN0eWxlZC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFFZSIsImZpbGUiOiJub24tZGVmYXVsdC1zdHlsZWQuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgeyBub25EZWZhdWx0U3R5bGVkIH0gZnJvbSAncGFja2FnZS1vbmUnXG5cbmxldCBTb21lQ29tcCA9IG5vbkRlZmF1bHRTdHlsZWQuZGl2KHsgY29sb3I6ICdob3RwaW5rJyB9KVxuIl19 */\\"
});"
`;
